src/main/java/uk/gov/pay/directdebit/payments/services/PaymentViewService.java

```diff
@@ -31,26 +38,42 @@ public PaymentViewResponse getPaymentViewResponse(PaymentViewSearchParams search
         return gatewayAccountDao.findByExternalId(searchParams.getGatewayExternalId())
                 .map(gatewayAccount -> {
                     PaymentViewSearchParams validatedSearchParams = paymentViewValidator.validateParams(searchParams);
-                    List<PaymentViewListResponse> viewListResponse = 
-                            getPaymentViewListResponse(paymentViewValidator.validateParams(searchParams));
+                    Long total = getTotal(validatedSearchParams);
+                    List<PaymentViewListResponse> viewListResponse =
+                            getPaymentViewListResponse(validatedSearchParams, gatewayAccount.getExternalId());
                     if (viewListResponse.size() == 0) {
                         throw new RecordsNotFoundException(format("Found no records with page size %s and display_size %s",
                                 validatedSearchParams.getPage(),
                                 validatedSearchParams.getDisplaySize()));
                     }
+                    ViewPaginationBuilder paginationBuilder = new ViewPaginationBuilder(validatedSearchParams, viewListResponse, uriInfo);
```

Comment:
I think you should remove the other exception as well. Also you could do 
```
if (total != 0) {
    getPaymentViewListResponse...
```
i.e. don't make do the search if you know there are no records